gatling-jms/src/main/scala/io/gatling/jms/client/JmsSendClient.scala

```diff
@@ -0,0 +1,157 @@
+/**
+ * Copyright 2011-2016 GatlingCorp (http://gatling.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gatling.jms.client
+
+import java.util.{ Hashtable => JHashtable }
+
+import io.gatling.core.config.Credentials
+import io.gatling.jms.protocol.JmsMessageMatcher
+import io.gatling.jms.request._
+
+import com.typesafe.scalalogging.StrictLogging
+import javax.jms._
+import javax.naming.{ Context, InitialContext }
+
+import scala.util.control.NonFatal
+
+/**
+ * Trivial JMS client, allows sending messages
+ */
+class JmsSendClient(
+  connectionFactoryName: String,
+  destination:           JmsDestination,
+  url:                   String,
+  credentials:           Option[Credentials],
+  anonymousConnect:      Boolean,
+  contextFactory:        String,
+  deliveryMode:          Int
+)

```

Comment:
Done. Put common staff into JmsClient and removed it from JmsReplyClient/JmsSendClient.